workflow-bot-app/src/test/java/com/symphony/bdk/workflow/SharedDataIntegrationTest.java

```diff
@@ -0,0 +1,40 @@
+package com.symphony.bdk.workflow;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.core.service.message.model.Message;
+import com.symphony.bdk.gen.api.model.V4Message;
+import com.symphony.bdk.workflow.swadl.SwadlParser;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+public class SharedDataIntegrationTest extends IntegrationTest {
+
+  @Test
+  @DisplayName("Share counter between process instances")
```

Comment:
Maybe we need another test where the counter is shared between 2 separate workflows.